tokens/usecase/pricing/coingecko/pricing_coingecko.go

```diff
@@ -0,0 +1,168 @@
+package coingeckopricing
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+	"github.com/osmosis-labs/sqs/domain"
+	"github.com/osmosis-labs/sqs/domain/cache"
+	"github.com/osmosis-labs/sqs/domain/mvc"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	cacheHitsCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "sqs_pricing_coingecko_cache_hits_total",
+			Help: "Total number of pricing coingecko cache hits",
+		},
+		[]string{"base", "quote"},
+	)
+	cacheMissesCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "sqs_pricing_coingecko_cache_misses_total",
+			Help: "Total number of pricing coingecko cache misses",
+		},
+		[]string{"base", "quote"},
+	)
+)
+
+// CoingeckoPriceGetterFn is a function type that fetches the price of a token from Coingecko.
+// We monkey-patch this function for testing purposes.
+type CoingeckoPriceGetterFn func(ctx context.Context, baseDenom string, coingeckoId string) (osmomath.BigDec, error)
+
+// DefaultCoingeckoPriceGetter is the default implementation of CoingeckoPriceGetterFn, which invokes GetPriceByCoingeckoId.
+var DefaultCoingeckoPriceGetter CoingeckoPriceGetterFn = nil
+
+// MockCoingeckoPriceGetter is a mock implementation of CoingeckoPriceGetterFn, which returns a fixed price of 1.
+var MockCoingeckoPriceGetter CoingeckoPriceGetterFn = func(ctx context.Context, baseDenom, coingeckoId string) (osmomath.BigDec, error) {
+	if coingeckoId == "" {
+		return osmomath.NewBigDec(0), nil
```

Comment:
nit: if we don't expect the results to be modified let's aim to define `osmomath` constants in the global space. These types take minor overhead for the initialization of internal buffers.

While negligible individually, these may add meaningful overhead to test times when accumulated